core/src/com/unciv/ui/worldscreen/unit/UnitActions.kt

```diff
@@ -563,15 +577,22 @@ object UnitActions {
             otherCiv.addNotification("[${unit.civInfo}] has stolen your territory!", unit.currentTile.position, unit.civInfo.civName, NotificationIcon.War)
     }
 
-    private fun addGoldPerGreatPersonUsage(civInfo: CivilizationInfo) {
-        val uniqueText = "Provides a sum of gold each time you spend a Great Person"
-        val cityWithMausoleum = civInfo.cities.firstOrNull { it.containsBuildingUnique(uniqueText) }
-                ?: return
-        val goldEarned = (100 * civInfo.gameInfo.gameParameters.gameSpeed.modifier).toInt()
-        civInfo.addGold(goldEarned)
-
-        val mausoleum = cityWithMausoleum.cityConstructions.getBuiltBuildings().first { it.uniques.contains(uniqueText) }
-        civInfo.addNotification("[${mausoleum.name}] has provided [$goldEarned] Gold!", cityWithMausoleum.location, NotificationIcon.Gold)
+    private fun addStatsPerGreatPersonUsage(civInfo: CivilizationInfo, unit: MapUnit) {
+        if (!unit.isGreatPerson()) return
+        
+        val gainedStats = Stats()
+        for (unique in civInfo.getMatchingUniques("Provides a sum of gold each time you spend a Great Person")) {
+            gainedStats.gold += (100 * civInfo.gameInfo.gameParameters.gameSpeed.modifier).toInt()
+        }
+        for (unique in civInfo.getMatchingUniques("[] whenever a great person is expended")) {
+            gainedStats.add(unique.stats)
+        }
+        
+        if (gainedStats.isEmpty()) return
+        
+        for (stat in gainedStats.toHashMap())
+            civInfo.addStat(stat.key, stat.value.toInt())
```

Comment:
Didn't you add a civInfo.addStats() function?

Reply:
Yes - but it was removed in a 'merge temp into master' commit or something. I can re-add it, of course